Update Haskell package set to Stackage Nightly 2021-03-06 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-03-12 20:00 UTC+1**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other.

## TODO

- [x] Drop pre-8.8.4 compiler versions.
- [x] Fix `cabal-install` builds with older compilers.

peti committed:
Stackage Nightly 2021-03-06

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/238cbc747dbd697d25278354a5e3db8d37afdd6c.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/952e6e1b2e861d21a062892e7dbc6873ac70300b.

roberth committed:
hercules-ci-agent: Fix and update build

roberth committed:
hci: init at 0.1.0

roberth committed:
haskellPackages: Add packages maintained by roberth

sternenseemann committed:
haskellPackages.cabal-install: install man page again

Previously, cabal-install had a custom Setup.hs which took care of
generating and installing the cabal(1) man page. While this file was a
bit of scary sight, it is certainly nice to have a man page properly
installed. For the 3.2.0.0 release they switched to the default setup
type again, so the man page isn't installed anymore. Fortunately the
cabal cli can generate the man page as well, so the override to add the
man page back is pretty simple.

The commit that introduced this is the following:
https://github.com/haskell/cabal/commit/91ac075930c87712eeada4305727a4fa651726e7
I actually made a mental note of this a few weeks ago already, but it
slipped my mind when we updated to cabal-install 3.2.0.0 two weeks ago
unfortunately.

cdepillabout committed:
Merge pull request #115347 from sternenseemann/cabal-install-man-pag

haskellPackages.cabal-install: install man page again

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/2b5683ae8200ab6d098e52220fa522298aae45b0.

maralorn committed:
haskell-language-server: 0.9.0 -> 1.0.0

Adapt overrides for newest hls version

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/9c425500b999f69cd4b38b158a40c64767859751.

markus1189 committed:
haskellPackages.amazonka: Unbreak by fetching patch

markus1189 committed:
Update pkgs/development/haskell-modules/configuration-common.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

markus1189 committed:
haskellPackages.fakedata: unbreak

cdepillabout committed:
Merge pull request #115484 from markus1189/haskellPackages-unbreak-amazonka

haskellPackages.amazonka: Unbreak by fetching patch

maralorn committed:
Merge pull request #115465 from maralorn/hls

[haskell-updates] haskell-language-server: 0.9.0 -> 1.0.0

markus1189 committed:
haskellPackages.fakedata-quickcheck: unbreak

markus1189 committed:
haskellPackages.string-random: unbreak

roberth committed:
Merge pull request #115330 from hercules-ci/hercules-ci-updates

Hercules CI updates

peti committed:
Stackage Nightly 2021-03-09

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/29bf5fcf6d3f386fe0984280a48b12233bdf7cfb.

cdepillabout committed:
Merge pull request #115509 from markus1189/haskellPackages-unbreak-fakedata

haskellPackages: Unbreak fakedata and fakedata-quickcheck and string-random

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/f151c5ab7f59585ac22e39de7b5c92f8c8eaecb9.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/0c2bc4b8bd8ba455526885fd837c9f4927b7f494.

maralorn commented:
This merge will include the long awaited 1.0 release of the haskell-language-server. :tada:

maralorn committed:
haskellPackages.hiedb: Make comment more explicit

peti committed:
Stackage Nightly 2021-03-12

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/07f8d034b42c78640aafac160da6828bcf806e0b.

maralorn committed:
haskellPackages.ghcide: Remove obsolete unmarkBroken and update test comment

peti committed:
git-annex: update SHA256 hash for new version 8.20210310

maralorn committed:
haskellPackages.shake-bench: Fix build

peti committed:
doctest-driver-gen: disable test suite to fix the build

maralorn committed:
haskellPackages.lsp-test: Remove obsolete dontCheck

lsp-test does not use hie anymore for tests

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/282dc91d4098800901ecf7d6d3549f9d75a5ddb3.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/282dc91d4098800901ecf7d6d3549f9d75a5ddb3.

peti committed:
feed: disable test suite to fix the build

The test suite does not compile.

peti committed:
hackage2nix: disable broken builds to avoid evaluation errors on Hydra

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/597fe9ca1cd918a6b1aa79cd0c3f53bc40317e04.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-13-gefcfbb2 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/6df578c11a9fe32710f2cdd32df89ee804b96135.

maralorn committed:
haskellPackages.prune-juice: Unbreak

maralorn committed:
haskellPackages: Add maralorn as shake-bench maintainer

It’s a dependency of hls development. But because it’s a benchmark dep I
wouldn‘t notice it breaking otherwise.

peti committed:
cabal-install: fix build with ghc-8.8.x

peti committed:
base64-bytestring: disable broken test suite

peti commented:
We'll merge https://github.com/NixOS/nixpkgs/pull/116096 next week.
